Foundations: Installations: Update VM instructions to VirtualBox 7, clarify some overall instructions

scheals opened the pull request:
<!-- Thank you for taking the time to contribute to The Odin Project. In order to get this pull request (PR) merged in a reasonable amount of time, you must complete this entire template. -->

## Because

A new version of VirtualBox has been released couple of months ago. It has some neat improvements and has gotten a rather significant UI facelift and as such our instructions currently do not reflect how the installation process looks like. Additionally, by default installing Xubuntu is done through a process called unattended installation which differs from what we had set up previously and comes with a quirk regarding `sudo` permissions that has to be alleviated after the install.

We've had and still are getting users installing VirtualBox 7 by accident and so far they have not reported any significant issues with using it, besides the aforementioned `sudo` quirk that we know how to fix and I have included in these instructions.

There are some very common issues that users run into that we provide support for on our Discord but could easily fit into our instructions, saving folks nerves and time related to their VM performance being bad due to misconfiguration, not knowing about having to disable Hyper-V or about how Google Chrome's scroll behaves differently from the scroll on Windows. Given that most of learners are not part of Discord, we can imagine a lot of people don't find support their need and might decide against continuing with the curriculum, decide to go against its recommendations or find bad advice on the Internet. Those common issues and their solutions have been included in the instructions.

Additionally, rarely we got users with macOS trying to install VirtualBox on their machines. To make matters worse, we've had users find other applications dealing with virtualization due to issues with running VirtualBox. I have included some more clarifications as who should be concerned with the OS installation instructions.


## This PR
- Updates our VM install instructions to how things are done on VirtualBox 7.
- Makes use of the unattended install function of VirtualBox 7.
- Gets rid of instructions for GuestAdditions install since it's done automatically by recommended unattended install config.
- Instructs learners how to fix `sudo` permission issues due to how unattended install function works.
- Introduces a FAQ centering on VM performance.
- Attempts to further clarify which users do not have to follow any of the OS installation instructions.

## Issue

Closes #24793

## Additional Information

This PR is WIP due to the fact that some decisions I've made I'm not exactly sure about:

- **Step 2.1**:  I suggested turning Python bindings support off during the VirtualBox install process due to it leading a user to see a page involving bolded red `WARNING:` message describing how one would need Python to make use of these bindings that how VirtualBox doesn't install Python. Given our experience with how users treat anything that starts with `warning` or looks like one (see: notes about unsandboxed installs) in my opinion it is best to just not expose them to it, because even when we provide an explanation, a question about it happens every once in a while. Still, explaining the warning would deal with the vast majority of questions about it.
- **Step 2.2.4**: There are two sets of instructions for enabling virtualization - one from Redhat that's the first set introduced and the other from 2N currently called alternative set of instructions. The first one is the more pronounced one in the current set of instructions, while the second one is somewhat deeper in them. The first one is more abstract and is written from Linux POV, while the second one is a lot more concrete, providing hotkeys to access BIOS for various machines and various names for Virtualization option in those machines. I think that the second one might be more helpful for our users at this point of the curriculum.
- **FAQ section/overall length of instructions**: Maybe it is too robust and not needed. I am not exactly sure. I thought the instructions will be significantly shorter after moving them to VirtualBox 7 since I was able to drop the entire GuestAdditions installation instructions and the installation process is overall shorter, with less windows to click through and input things in. That doesn't seem to have panned out but I still think the instructions are *much, much* less complex now and FAQ's inclusion is going to help lots of folks, so even if we don't gain on length front, we gain on legibility, reduce friction for newcomers and we provide them with answers to common headaches that might dissuade them from using a VM.
- **Attempts at clarifying who the lesson/instructions are for**: I have changed the very first two paragraphs to make it even more clear who should follow the lesson and/or the OS installation instructions. Rarely we get people who are asking about having to set up a dual boot after setting up a VM or we get people who are trying to set up a VM or a dual boot on their macOS systems. Hopefully this reduces the confusion but maybe the instructions are clear enough as of now and we just have to support those who are confused about them nonetheless.
- **Style**: due to how images are styled, the icons don't fit well in the text - the stylesheet makes them `block` instead of `inline`. I have made sure to apply inline styling to alleviate this so the instructions read normally but it goes against the Style Guide pretty much. 

Also, I have not removed images used for previous instructions due to reusing some of them + I have to make a separate PR for adding pictures so I can host them on the CDN, so after that's done I'll edit this PR to use Statically links and apply that image change. Not sure if my naming convention is alright, the style guide mentions using just numbers like `00`, `01`, `02` and so on so am not exactly sure if it can be the case that they have something prepended to them as well. Anywho, whenever I get an OK about the PR, I'll accommodate the image names and unify them, both reused and new.

Unrelated:
- Headings and `<code>` blocks don't play nice styling-wise - in `h3` headers, the `code` blocks inside `a` elements don't have correct margins, padding & borders. The `h4` headers seem to reset the colour applied to `code` blocks. I have this on my radar and will create an issue for this on the main repo somewhat soon.


## Pull Request Requirements
<!-- Replace the whitespace between the square brackets with an 'x', e.g. [x]. After you create the PR, they will become checkboxes that you can click on. -->
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

xandora reviewed (changes requested):
Looking good! Couple things to address, mostly the images being added to the CDN, they should all be marked by a comment. Nice work :)

scheals commented on foundations/installations/installations.md:
```diff
@@ -28,185 +28,180 @@ Installing a VM is a simple process. This guide uses Oracle's VirtualBox program
 #### IMPORTANT
 
-Once you have completed these instructions, **you are expected to work entirely in the VM.** Maximize the window, add more virtual monitors if you have them, fire up the Internet Browser in the **Whisker Menu** ![The Whisker Menu Icon](https://cdn.statically.io/gh/TheOdinProject/curriculum/5d27ddb08c8cf3c553537deb6156a5c7f7aa1bac/foundations/installations/prerequisites/imgs/whisker_menu_icon.png) ...
+Once you have completed these instructions, **you are expected to work entirely in the VM.** Maximize the window, add more virtual monitors if you have them, fire up the Internet Browser in the **Whisker Menu** <img src="https://cdn.statically.io/gh/TheOdinProject/curriculum/5d27ddb08c8cf3c553537deb6156a5c7f7aa1bac/foundations/installations/prerequisites/imgs/whisker_menu_icon.png" alt="The blue- ...
```
Yep, otherwise it just breaks the flow, examples of how it currently looks on site:
![image](https://user-images.githubusercontent.com/95084627/211091092-5a603d9c-2ad6-4be0-9435-f347b1c9ab63.png)
![image](https://user-images.githubusercontent.com/95084627/211091101-867fe7ea-2c66-423c-98f2-53336b5857d4.png)

In this case, it ain't terrible but it is not good either. Inline looks a lot better & the FAQ section with the VBox icons was awful with block styling.

xandora replied:
Sounds good :)

scheals commented on foundations/installations/installations.md:
```diff
@@ -28,185 +28,180 @@ Installing a VM is a simple process. This guide uses Oracle's VirtualBox program
+#### Step 3.1: Navigate to Users and Group
 
-You should now see the ISO on the Disk Selector screen. Click it and hit the **Choose** button at the bottom.
+First, click on the <img src="https://cdn.statically.io/gh/TheOdinProject/curriculum/5d27ddb08c8cf3c553537deb6156a5c7f7aa1bac/foundations/installations/prerequisites/imgs/whisker_menu_icon.png" alt="The blue-white rodent Whisker Menu Icon" style="display: inline; margin: auto;"> in top left of your window, afterwards type in `Users` and you should see `Users and Groups` show up. Click on it.
```
Yes, all the inline stylings for are to make sure icons flow neatly with the text.

scheals commented on foundations/installations/installations.md:
```diff
@@ -28,185 +28,180 @@ Installing a VM is a simple process. This guide uses Oracle's VirtualBox program
+    -   When running the VM, minimize your Windows activity. You probably want to be plugged into power as well if you're using a laptop.
+    -   Make sure your processors are set to only 2 and the Memory you've given your VM is at most half of your total RAM but at least 2GB. If you can't spare the 2GB, dual boot.
+    -   If videos lag in the VM, make sure to max out Video Memory to whatever you can or alternatively play them on your Windows if your machine can handle it. Disable 3D Acceleration if you have it enabled.
+    -   Do look out for the <img src="./installations/imgs/17_turtle.webp" alt="Icon with a green turtle and a V" style="display: inline; margin: auto;"> icon in bottom right of your VM window, if you have it, here's a [VirtualBox forum thread on how to get rid of it](https://forums.virtualbox.org/viewtopic.php?f=25&t=99390). You want to have the <img src="./installations/imgs/18_vboxV.webp" alt= ...
```
Is it a problem? I just grabbed them off the link Discord provides. I assume I can convert them.

xandora replied:
Might be? They don't seem to preview on GitHub at all. It's possible that if we have, for whatever reason, someone who browses the curriculum through GitHub alone, they might miss out on the image content.

scheals replied:
Alright. I'll look into converting them and the PR to add images will have everything in `.png` or whatever the rest is.

scheals commented on foundations/installations/installations.md:
```diff
@@ -28,185 +28,180 @@ Installing a VM is a simple process. This guide uses Oracle's VirtualBox program
+    -   When running the VM, minimize your Windows activity. You probably want to be plugged into power as well if you're using a laptop.
+    -   Make sure your processors are set to only 2 and the Memory you've given your VM is at most half of your total RAM but at least 2GB. If you can't spare the 2GB, dual boot.
+    -   If videos lag in the VM, make sure to max out Video Memory to whatever you can or alternatively play them on your Windows if your machine can handle it. Disable 3D Acceleration if you have it enabled.
+    -   Do look out for the <img src="./installations/imgs/17_turtle.webp" alt="Icon with a green turtle and a V" style="display: inline; margin: auto;"> icon in bottom right of your VM window, if you have it, here's a [VirtualBox forum thread on how to get rid of it](https://forums.virtualbox.org/viewtopic.php?f=25&t=99390). You want to have the <img src="./installations/imgs/18_vboxV.webp" alt= ...
```
Same as above.

xandora replied:
Same as above. :D

scheals commented on foundations/installations/installations.md:
```diff
@@ -28,185 +28,180 @@ Installing a VM is a simple process. This guide uses Oracle's VirtualBox program
+With `sudo` selected, click **Preferences** and in the window that shows up tick your user's name like this:
 
-### Step 3: Install and Enable Guest Additions
+   ![No need to touch anything else.](./installations/imgs/13_sudo_properties.png)
```
Aight, proposed PR https://github.com/TheOdinProject/curriculum/pull/25006 , so if any changes are needed, lmk there :D

scheals commented:
Moved imwheel instructions after the performances ones are and added a note about the text editor used because it was a common issue with users following the askubuntu answer.

xandora reviewed (approved):
looks good to me.

xandora reviewed (approved):
👍

xandora commented:
Let's goooooooooooooooooo!
